plugwise_usb/nodes/__init__.py

```diff
@@ -536,57 +554,39 @@ def _node_info_update_state(
 
     async def is_online(self) -> bool:
         """Check if node is currently online."""
-        try:
-            ping_response: NodePingResponse | None = await self._send(
-                NodePingRequest(
-                    self._mac_in_bytes, retries=1
-                )
-            )
-        except StickError:
-            _LOGGER.warning(
-                "StickError for is_online() for %s",
-                self.mac
-            )
-            await self._available_update_state(False)
-            return False
-        except NodeError:
-            _LOGGER.warning(
-                "NodeError for is_online() for %s",
-                self.mac
-            )
-            await self._available_update_state(False)
-            return False
-
-        if ping_response is None:
-            _LOGGER.info(
-                "No response to ping for %s",
-                self.mac
-            )
-            await self._available_update_state(False)
-            return False
-        await self.ping_update(ping_response)
-        return True
+        result = await self.ping_update()
+        return result is not None
 
     async def ping_update(
         self, ping_response: NodePingResponse | None = None, retries: int = 1
     ) -> NetworkStatistics | None:
         """Update ping statistics."""
         if ping_response is None:
-            ping_response = await self._send(
-                NodePingRequest(
-                    self._mac_in_bytes, retries
+            try:
+                ping_response: NodePingResponse | None = await self._send(
+                    NodePingRequest(
+                        self._mac_in_bytes, retries=1
+                    )
                 )
-            )
+            except (StickError, NodeError) as e:
+                _LOGGER.warning(
+                    "%s for ping_update() for %s",
+                    str(e),
+                    self.mac
+                )
+                await self._available_update_state(False)
+                return False
         if ping_response is None:
             await self._available_update_state(False)
-            return None
+            return False
```

Comment:
The ping_update function (just like any other *_update function) should return the typed result. So in this case the populated `NetworkStatistics` dataclass object, of None if it failed.